app/src/main/java/edu/usf/sas/pal/muser/model/UiEvent.kt

```diff
@@ -0,0 +1,18 @@
+package edu.usf.sas.pal.muser.model
+
+/**
+ *  A model class that stores different types of events actions performed on the song and implements
+ *  Event interface.
+ */
+data class UiEvent
+
+/**
+ * [uiEventType] - The enum object for the type of action performed.
+ */
+(val uiEventType: UiEventType,
+ override val currentTimeMs: Long,
+ override val nanoTime: Long,
+ override val startTime: Long,
+ override val elapsedTime: Long,
+ override val song: SongData
```

Comment:
`song` needs to be nullable here in case we have UI events that don't include a song